pkg/bgpv1/manager/route_policy_reconciler.go

```diff
@@ -186,8 +192,27 @@ func (r *RoutePolicyReconciler) pathAttributesToPolicy(attrs v2alpha1api.CiliumB
 	}
 
 	switch attrs.SelectorType {
+	case v2alpha1api.CiliumPodIPPoolSelectorName:
+		localPools := r.populateLocalPools(params.CiliumNode)
+		for _, pool := range r.podPoolStore.List() {
+			if attrs.Selector != nil && !labelSelector.Matches(labels.Set(pool.Labels)) {
+				continue
+			}
+			// only advertise pool cidrs that have been allocated to the local node.
```

Comment:
Comment says 'advertise' but technically this is not to do any advertisement. It only creates egress policy.

Reply:
Thanks for catching this, I'll update.